### PR TITLE
Firefox on Darwin

### LINKED ISSUE #30285: Support Firefox on darwin?
Currently, all the Firefox derivations are Linux-only. I would love to have Firefox managed by Nix on darwin.

I would be interested in taking this on myself, but the Firefox derivations seem fairly complicated and I have failed at much more basic Nix stuff. So, if anyone has at least a pointer on how to go about it, I would be very happy. Also, I’m personally only interested in 57/Quantum, in case that makes things easier or harder.

### PR DESCRIPTION
**Pull request by matthewbauer**
###### Motivation for this change
This is my working attempt to get Firefox working on Darwin/macOS. This will fix #30285.

Unfortunately this is a mass rebuild because of 0078e82ac240d88e3a20afcbf134904cbf920893. Unless there is legitimate reason for leaving it out I will probably just push that commit to staging. Anyway I am still waiting on things here but should be ready to go in a day or two.

### CODE REVIEW

**Comment by LnL7** [comment]
I don't have a problem with adding GUI applications. That said they generally seem to be much harder to maintain so if it's too much of a burden we shouldn't keep it in the release blockers, but let's just see how it goes.

File: pkgs/development/tools/xcbuild/wrapper.nix [hunk 1]
```diff
@@ -36,22 +37,22 @@ let
...
+         -p | -print-path | --print-path) echo @DEVELOPER_DIR@ ;;
          --install) ;; # noop
     esac
     shift
 done
   '';
 
   xcrun = writeText "xcrun" ''
-#!/usr/bin/env sh
+#!${bash}/bin/sh
```
[the remark this answers is not in the document]
Comment (matthewbauer) [reply pkgs/development/tools/xcbuild/wrapper.nix]: Thanks! Can change to that.

**Comment by copumpkin** [comment]
Very exciting, thanks again!

**Comment by vcunat** [comment]
```
Testing on x86_64-linux:
error: while querying the derivation named 'conkeror-1.0.4':
while evaluating the attribute 'buildCommand' of the derivation 'conkeror-1.0.4' at /home/admin/nix/nixpkgs-x/pkgs/stdenv/generic/make-derivation.nix:179:11:
attribute 'execdir' missing, at /home/admin/nix/nixpkgs-x/pkgs/applications/networking/browsers/firefox/wrapper.nix:109:32
```

**Comment by vcunat** [comment]
There's also `midori` and `vimb`, maybe more.  I didn't really look into these changes, but perhaps it should just use a default if the attribute isn't defined?

**Comment by matthewbauer** [comment]
Yeah it can default to /bin. Sorry forgot about that.

**Pull request merged**